src/script/standard.cpp

```diff
@@ -110,7 +110,7 @@ bool Solver(const CScript& scriptPubKey, txnouttype& typeRet, std::vector<std::v
             // Template matching opcodes:
             if (opcode2 == OP_PUBKEYS)
             {
-                while (vch1.size() >= 33 && vch1.size() <= 65)
+                while (CPubKey::ValidSize(vch1))
```

Comment:
This (and the change below) OTOH is *not* a consensus change, only a standard rule change. Technically this could change the behaviour of existing (buggy) ecosystem software, but I don't think it's a problem.